### ISSUE TITLE
how to configure the datset

### ISSUE THREAD

**Issue by brianhu0716** [lhoyer/MIC#31]
Hi, authors

I have a question about using MIC in domain adaption object detection task. Say, I have a data set which is a subset from cityscapes data set just as that used in your paper. And I organize it as following

data

    •   cityscapes (source)
      
      - train image folder (with labeling)
      
      - val image folder (with labeling)
      
    • foggy_cityscapes (target)
      
      - train image folder (without labeling)

      - val image folder (with labeling)

the problem is when I revising the corresponding path in DatasetCatalog class, I notice that it requires user to specify the annotation path even the target training data set is under consideration.

If I’m not misunderstanding to the method described in your paper, the target samples should not have any ground truth and the pseudo-label is generated by EMA-teacher model. The question is  why do we still need to give annotations to the target training set? 

Could you please give me some insight about how to configure the paths to fit the data I have? Thank you

**Comment by lhoyer** [comment]
Hi @brianhu0716,

Thank you for your interest in our work.  The target domain annotations are never used during training and are just a legacy artifact as discussed in #11 and https://github.com/lhoyer/DAFormer/issues/34.

Best,
Lukas

**Comment by brianhu0716** [comment]
Hi, Ihoyer

yes, I've read the related issues, but it seems that no one point out how to set the proper path to the key-value pair in DatasetCatalog

        "foggy_cityscapes_fine_instanceonly_seg_train_cocostyle": {
            "img_dir": "foggy_cityspace/train",
            "ann_file": **None** # TODO: none for the reason that there is no label in target training set
        },

In the current stage, I set the path as None, but it seems to have some issues when preparing the dataset. So, am I right to do the revision above? or the path of target training set should be initialized in another way to avoid the errors reported in the subsequent steps. thank you

**Comment by krumo** [comment]
Hi @brianhu0716,

As Lukas points out, due to the compatibility issue you still have to pass an annotation file when you train on domain adaptive object detection. I would refer you to discussion in https://github.com/lhoyer/MIC/issues/21 about how I avoid using target domain annotations in the codebase. 

For adding a new target domain dataset, I would suggest you create a COCO format pseudo annotation file where each image only has one pseudo bbox annotation like what I do in https://github.com/lhoyer/MIC/blob/1aa2f5ea6aed4b5601d5106464302255470b86af/det/maskrcnn_benchmark/data/datasets/coco.py#L76.

Best,
Haoran

**Comment by brianhu0716** [comment]
@krumo 

thaks for replying with an explicit explanation

so it seems that the compromised solution to this problem is generating a pseudo-annotations file for target training set, and the meaningless bounding boxes information are never used in the following training steps. 

I will try it soon and report the result if there still exist error. Please correct me if I figure your description in a wrong way. 

thank you!
